core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,562 @@
+                                type=lambda x: existing_file(x, create_dir=True, shouldnt_exist=True),
+                                help="temporary directory for files (default: ./tmp)")
+    optional_group.add_argument("--seed", metavar="N", type=int,
+                                help="seed for reproducibility")
+    optional_group.add_argument("-n", "--nperm", metavar="N", type=int, default=1000,
+                                help="number of permutations for p-value estimation (default: 1000)")
+    optional_group.add_argument("--alt-allele", dest="alt_allele", choices=["Tumor_Seq_Allele1", "Tumor_Seq_Allele2"],
+                                help="Manually set alternative allele")
+    optional_group.add_argument("--annotate", metavar="annotate.csv", type=existing_file,  # Location relative to repo
+                                default=osp.normpath(osp.join(osp.dirname(__file__), 
+                                                     "../resources/mutational_signatures_annotation.csv")),
+                                help="Path to signature annotation file (default: "
+                                     "../resources/mutational_signatures_annotation.csv)")
+    args = parser.parse_args()
+    # Check some conditionally required arguments
+    arg_conditionals(args)
+    return args
+
+
+def existing_file(path, create_dir=False, shouldnt_exist=False):
+    """Check if file or directory exists
+    
+    :param path: file or directory path 
+    :type path: str
+    :param create_dir: create directory if non-existent
+    :param shouldnt_exist: raise error if exists
+    :return: file or directory path
+    :rtype: str
+    """
+    if not os.path.exists(path):
+        if not create_dir:
+            raise argparse.ArgumentTypeError(f"{path} does not exist.")
+        os.mkdir(path)
+        return path
+    if shouldnt_exist:
+        raise argparse.ArgumentTypeError(f"{path} should not exist before running the script.")
+    return path
+
+
+def implemented_genome(genome: str) -> str:
+    """Check if genome has been implemented. Ambiguous options default to human genome.
+    
+    :param genome: genome
+    :return: NCBI build
+    """
+    ncbi_build_options = {
+        "19": "GRCh37", "hg19": "GRCh37", "grch37": "GRCh37", "37": "GRCh37", "GRCh37": "GRCh37",
+        "grch38": "GRCh38", "38": "GRCh38", "GRCh38": "GRCh38",
+        "mm9": "GRCm37", "9": "GRCm37", "grcm37": "GRCm37",
+        "mm10": "GRCm38", "10": "GRCm38", "grcm38": "GRCm38"
+    }
+    ncbi_build = ncbi_build_options.get(genome)
+    if ncbi_build is None:
+        raise argparse.ArgumentTypeError(f"Genome: `{genome}` has not been implemented. "
+                                         f"The following genomes are implemented: {set(ncbi_build_options.values())}")
+    return ncbi_build
+        
+        
+def arg_conditionals(args: argparse.Namespace):
```

Comment:
To my knowledge, there are no options in `ArgParse` for the conditionals in this function:
- 1 or more of these options is needed
- If option1 is chosen, option2 is required, otherwise not